service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/TemporaryHearingJudgeContentProviderTest.java

```diff
@@ -0,0 +1,109 @@
+@ContextConfiguration(classes = {TemporaryHearingJudgeContentProvider.class})
+class TemporaryHearingJudgeContentProviderTest extends AbstractEmailContentProviderTest {
+    private static final LocalDateTime NOW = LocalDateTime.now();
+
+    @Autowired
+    private TemporaryHearingJudgeContentProvider temporaryHearingJudgeContentProvider;
+
+    @Test
+    void shouldBuildTemporaryHearingJudgeTemplateWithExpectedParameters() {
+        CaseData caseData = CaseData.builder()
+            .id(12345L)
+            .respondents1(List.of(ElementUtils.element(Respondent.builder()
+                .party(RespondentParty.builder()
+                    .firstName("Mark")
+                    .lastName("Watson")
+                    .build())
+                .build())))
+            .familyManCaseNumber("123")
+            .allocatedJudge(Judge.builder()
+                .judgeLastName("Watson")
+                .judgeTitle(HIS_HONOUR_JUDGE)
+                .build())
+            .build();
+
+        HearingBooking hearingBooking = HearingBooking.builder()
+            .type(CASE_MANAGEMENT)
+            .startDate(NOW)
+            .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                .judgeTitle(HER_HONOUR_JUDGE)
+                .judgeLastName("Davidson")
+                .build())
+            .build();
+
+        assertThat(temporaryHearingJudgeContentProvider.buildNotificationParameters(caseData, hearingBooking))
+            .isEqualToComparingFieldByField(getExpectedNotificationParameters());
+    }
+
+    @Test
+    void shouldBuildPartialTemporaryHearingJudgeTemplateWhenMissingOptionalCaseData() {
+        CaseData caseData = CaseData.builder()
+            .id(12345L)
+            .familyManCaseNumber("123")
+            .build();
+
+        HearingBooking hearingBooking = HearingBooking.builder()
+            .type(CASE_MANAGEMENT)
+            .startDate(NOW)
+            .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                .judgeTitle(HER_HONOUR_JUDGE)
+                .judgeLastName("Davidson")
+                .build())
+            .build();
+
+        TemporaryHearingJudgeTemplate partiallyCompleteTemplate =
+            temporaryHearingJudgeContentProvider.buildNotificationParameters(caseData, hearingBooking);
+
+        String hearingStartDate = formatLocalDateToString(NOW.toLocalDate(), FormatStyle.MEDIUM);
+
+        assertThat(partiallyCompleteTemplate.getCallout()).isEqualTo(
+            String.format("123, hearing %s", hearingStartDate)
+        );
+
+        assertThat(partiallyCompleteTemplate.getHasAllocatedJudge()).isEqualTo(NO.getValue());
+        assertThat(partiallyCompleteTemplate.getAllocatedJudgeName()).isEmpty();
+        assertThat(partiallyCompleteTemplate.getAllocatedJudgeTitle()).isEmpty();
+    }
+
+    private TemporaryHearingJudgeTemplate getExpectedNotificationParameters() {
+        TemporaryHearingJudgeTemplate allocatedJudgeTemplate = new TemporaryHearingJudgeTemplate();
```

Comment:
bit pedantic but can we name the variable something else, we already use allocatedJudgeTemplate elsewhere in the code to refer to allocated judge, and this is the temporary judge not the allocated judge so it might be confusing at a glance